ansible/scylla-migrator.yml

```diff
@@ -0,0 +1,250 @@
+
+    - name: install awscli v2
+      become: true
+      shell: "sudo {{ home_dir }}/aws/install --update"
+
+    - name: Delete aws zip file
+      ansible.builtin.file:
+        state: absent
+        path: "{{ home_dir }}/awscliv2.zip"
+
+    - name: Delete aws install directory
+      ansible.builtin.file:
+        state: absent
+        path: "{{ home_dir }}/aws"
+
+    - name: Download sbt
+      ansible.builtin.get_url:
+        url: https://github.com/coursier/coursier/releases/latest/download/cs-x86_64-pc-linux.gz
+        dest: "{{ home_dir }}"
+
+    - name: unarchive sbt
+      shell: gzip -d cs-x86_64-pc-linux.gz
+
+    - name: Delete /tmp/cs-x86_64-pc-linux.gz
+      ansible.builtin.file:
+        state: absent
+        path: "{{ home_dir }}/cs-x86_64-pc-linux.gz"
+
+    - name: rename cs-x86_64-pc-linux to cs
+      become: yes
+      shell: mv cs-x86_64-pc-linux cs
+  
+    - name: Change permission on sbt
+      file:
+        path: cs
+        state: file
+        mode: 0755
+
+    - name: setup sbt
+      shell: "{{ home_dir }}/cs setup -y"
+
+    - name: Download spark
+      get_url: 
+        url: https://archive.apache.org/dist/spark/spark-2.4.8/spark-2.4.8-bin-hadoop2.7.tgz 
+        dest: "{{ home_dir }}/spark-2.4.8-bin-hadoop2.7.tgz"
+
+    - name: Extract spark
+      ansible.builtin.unarchive:
+        src: "spark-2.4.8-bin-hadoop2.7.tgz"
+        dest: "{{ home_dir }}"
+        remote_src: yes
+
+    - name: Empty spark home
+      become: true
+      ansible.builtin.file:
+        state: absent
+        path: "{{ spark_home }}"
+
+    - name: Move spark to opt/spark
+      become: true
+      command: "sudo mv spark-2.4.8-bin-hadoop2.7 {{ spark_home }}"
+
+    - name: Set JAVA_HOME
+      ansible.builtin.lineinfile:
+        path: "{{ home_dir }}/.profile"
+        regexp: '^JAVA_HOME='
+        line: export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
+
+    - name: Add JAVA_HOME to PATH
+      ansible.builtin.lineinfile:
+        path: "{{ home_dir }}/.profile"
+        regexp: '^:$JAVA_HOME'
+        line: export PATH=$PATH:$JAVA_HOME
+
+    - name: Set SPARK_HOME
+      ansible.builtin.lineinfile:
+        path: "{{ home_dir }}/.profile"
+        regexp: '^:SPARK_HOME'
+        line: "export SPARK_HOME={{ spark_home }}"
+
+    - name: Add SPARK_HOME/bin to path
+      ansible.builtin.lineinfile:
+        path: "{{ home_dir }}/.profile"
+        regexp: '^:$SPARK_HOME/bin'
+        line: export PATH=$PATH:$SPARK_HOME/bin
+
+    - name: Add SPARK_HOME/sbin to path
+      ansible.builtin.lineinfile:
+        path: "{{ home_dir }}/.profile"
+        regexp: '^:$SPARK_HOME/sbin'
+        line: export PATH=$PATH:$SPARK_HOME/sbin
+
+    - name: Generate an OpenSSH keypair with the default values (4096 bits, rsa)
+      community.crypto.openssh_keypair:
+        path: "{{ home_dir }}/.ssh/id_ssh_rsa"
+        state: present
+      register: rsa_key
+
+    - name: Allow passwordless SSH between all hosts
```

Comment:
shouldn't this be done outside? I mean for this playbook to run it is expected that you have passwordless ssh ...

Reply:
@pdbossman once you will check if this is really needed and ev. remove it, I will merge ...